File: DMOJ/ccc15j2.py
input_str = input()

list_inputt = list(input_str)

happy_counter = 0
sad_counter = 0

for i in range(0, len(list_inputt)):
    if list_inputt[i] == ':' and list_inputt[i+1] == '-':
        if list_inputt[i + 2] == ')':
            happy_counter += 1
        elif list_inputt[i+2] == '(':
            sad_counter += 1

if happy_counter > sad_counter:
    print("happy")
elif happy_counter < sad_counter:
    print("sad")
elif happy_counter == sad_counter and happy_counter != 0:
    print("unsure")
else:
    print("none")

File: DMOJ/dmoj.ca-downloaded-submissions/ccc15j2.cpp
#include <iostream>
#include <string> 

using namespace std; 

string findNumber(string astring){
    int happyCounter = 0;
    int sadCounter = 0;
    if (astring.length() > 2){
        for (int i = 0; i < astring.length() - 3; i++)
        {
            if (astring.substr(i, 3) == ":-)")
            {
                happyCounter ++; 
            }
            else if (astring.substr(i,3) == ":-(")
            {
                sadCounter ++; 
            }
        }        
    } 


    if (happyCounter > sadCounter){
        return "happy";
    }else if (happyCounter < sadCounter){
        return "sad"; 
    }else if (happyCounter == 0 && sadCounter == 0){
        return "none"; 
    }else if (happyCounter == sadCounter){
        return "unsure";
    }
    else{
        return "none";
    }
}

int main(){
    string message;
    getline(cin,message); 
    cout << findNumber(message) << endl; 
    return 0;
}